src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -485,15 +682,377 @@ This idiom safely accesses `person.spouse.name` if `spouse` exists; otherwise, i
+UPDATE province:bc      SET next = [city:vancouver, city:victoria];
+RELATE state:california ->has->    [city:los_angeles, city:san_francisco];
+RELATE state:texas      ->has->    [city:houston, city:dallas];
+RELATE province:bc      ->has->    [city:vancouver, city:victoria];
+RELATE province:ontario ->has->    [city:toronto, city:ottawa];
+```
+
+Before version `2.1.0`, traversing each of these paths could only be done manually, requiring a good deal of typing and knowing the exact depth to traverse.
+
+Here is an example using record links:
+
+```surql
+SELECT 
+	next AS countries,
+	next.next AS states_provinces,
+	next.next.next AS cities
+FROM planet:earth;
+```
+
+```surql title="Response"
+[
+	{
+		cities: [
+			[
+				[
+					city:los_angeles,
+					city:san_francisco
+				],
+				[
+					city:houston,
+					city:dallas
+				]
+			],
+			[
+				[
+					city:toronto,
+					city:ottawa
+				],
+				[
+					city:vancouver,
+					city:victoria
+				]
+			]
+		],
+		countries: [
+			country:us,
+			country:canada
+		],
+		states_provinces: [
+			[
+				state:california,
+				state:texas
+			],
+			[
+				province:ontario,
+				province:bc
+			]
+		]
+	}
+]
+```
+
+And here is an example using graph links.
+
+```surql
+SELECT 
+	-- Show all `country` records located at `out`
+	->has->country AS countries,
+	-- Show all `province` or `state` records located at `out`	
+	->has->country->has->(province, state) AS state_provinces,
+	-- Or use (?) to show any type of record located at `out`
+	->has->(?)->has->(?)->has->(?) AS cities
+FROM planet:earth;
+```
+
+```surql title="Output"
+[
+	{
+		cities: [
+			city:toronto,
+			city:ottawa,
+			city:vancouver,
+			city:victoria,
+			city:dallas,
+			city:houston,
+			city:los_angeles,
+			city:san_francisco
+		],
+		countries: [
+			country:canada,
+			country:us
+		],
+		state_provinces: [
+			province:ontario,
+			province:bc,
+			state:texas,
+			state:california
+		]
+	}
+]
+```
+
+#### Basics of recursive paths
+
+Using a recursive path allows you to instead set the number of steps to follow instead of manually typing. A recursive path is made by isolating `{}` braces in between two dots, inside which the number of steps is indicated.
+
+```surql
+-- Two steps down the record links at the `next` field
+planet:earth.{2}.next;
+-- Two steps down the `has` graph relation
+planet:earth.{2}->has->(?);
+```
+
+```surql title="Output"
+[
+	state:california,
+	state:texas,
+	province:ontario,
+	province:bc
+]
+```
+
+The number of steps can be any integer from 1 to 256.
+
+```surql
+-- 'Found 0 for bound but expected at least 1.'
+planet:earth.{0}->has->(?);
+-- 'Found 500 for bound but expected 256 at most.'
+planet:earth.{500}->has->(?);
+```
+
+A range can be inserted into the braces to indicate a desired minimum and maximum depth. 
+
+```surql
+-- Returns [] because no 4th-level relations exist
+planet:earth.{4}->has->(?);
+-- Returns `city` records located at depth 3
+planet:earth.{1..4}->has->(?);
```

Comment:
I know it's only a PR for the documentation but it should support all ranges syntax like `planet:earth.{1..=4}->has->(?);` as well. It doesn't at the moment so the documentation is right but I wanted to add it.

Reply:
@Ekwuno @Dhghomon taken into account too ? 👆